helix-core/src/main/java/org/apache/helix/monitoring/mbeans/InstanceMonitor.java

```diff
@@ -67,14 +74,12 @@ public String metricName() {
 
   private List<String> _tags;
 
-  // Counters
-  private SimpleDynamicMetric<Long> _totalMessagedReceivedCounter;
-
-  // Gauges
-  private SimpleDynamicMetric<Long> _enabledStatusGauge;
-  private SimpleDynamicMetric<Long> _disabledPartitionsGauge;
-  private SimpleDynamicMetric<Long> _onlineStatusGauge;
-  private SimpleDynamicMetric<Double> _maxCapacityUsageGauge;
+  // Capacity Gauges
+  private final Map<String, SimpleDynamicMetric<Long>> _capacityMetricsMap;
```

Comment:
I like the design. But it could be confusing why _capacityMetricsMap is not part of the _longMetricsMap.
Can we make it more differentiate?

Reply:
Currently I don't have a better name for this. Maybe we can add clearer comments for it?

Reply:
Sure, it should work.
Or call _capacityMetricsMap -> _dynamicCapacityMetricsMap to emphasis that it is changeable?

Reply:
I also considered _dynamicCapacityMetricsMap / _mutableCapacityMetricsMap. I will change the name and also add more comments about the mutable map.